services/src/main/java/org/keycloak/protocol/oid4vc/issuance/mappers/OID4VCIssuedAtTimeClaimMapper.java

```diff
@@ -0,0 +1,145 @@
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * Map issuance date to the credential, under the default claim name "iat"
+ * <p>
+ * subjectProperty can be used to change the claim name.
+ * <p>
+ * Source of the information can either be computed, or read from the VerifiableCredential object
+ * bearing other claims. Default is the value in the verifiable credential.
+ * <p>
+ * We will use the java.time.temporal.ChronoUnit enum values to help flatten down the time.
+ *
+ * @author <a href="mailto:francis.pouatcha@adorsys.com">Francis Pouatcha</a>
+ */
+public class OID4VCIssuedAtTimeClaimMapper extends OID4VCMapper {
+
+    public static final String MAPPER_ID = "oid4vc-issued-at-time-claim-mapper";
+
+    // Omit value if defaults to "iat"
+    public static final String SUBJECT_PROPERTY_CONFIG_KEY = "subjectProperty";
+
+    // We will use the java.time.temporal.ChronoUnit enum values to help flatten down the time.
+    // Omit property if no truncation.
+    public static final String TRUNCATE_TO_TIME_UNIT_KEY = "truncateToTimeUnit";
+
+    // Time computed (COMPUTE) or taken from the verifiable credential (VC).
+    // Defaults to VC. Falls back to COMPUTE.
+    public static final String VALUE_SOURCE = "valueSource";
+
+    private static final List<ProviderConfigProperty> CONFIG_PROPERTIES = new ArrayList<>();
+
+    static {
+        ProviderConfigProperty subjectPropertyNameConfig = new ProviderConfigProperty();
+        subjectPropertyNameConfig.setName(SUBJECT_PROPERTY_CONFIG_KEY);
+        subjectPropertyNameConfig.setLabel("Time Claim Name");
+        subjectPropertyNameConfig.setHelpText("Name of this time claim. Default is iat");
+        subjectPropertyNameConfig.setType(ProviderConfigProperty.STRING_TYPE);
+        subjectPropertyNameConfig.setDefaultValue("iat");
+        CONFIG_PROPERTIES.add(subjectPropertyNameConfig);
+
+        ProviderConfigProperty truncateToTimeUnit = new ProviderConfigProperty();
+        truncateToTimeUnit.setName(TRUNCATE_TO_TIME_UNIT_KEY);
+        truncateToTimeUnit.setLabel("Truncate To Time Unit");
+        truncateToTimeUnit.setHelpText("Truncate time to the first second of the MINUTES, HOURS, HALF_DAYS, DAYS, WEEKS, MONTHS or YEARS. Such as to prevent correlation of credentials based on this time value.");
+        truncateToTimeUnit.setType(ProviderConfigProperty.LIST_TYPE);
+        truncateToTimeUnit.setOptions(List.of("MINUTES", "HOURS", "HALF_DAYS", "DAYS", "WEEKS", "MONTHS", "YEARS"));
+        CONFIG_PROPERTIES.add(truncateToTimeUnit);
+
+        ProviderConfigProperty valueSource = new ProviderConfigProperty();
+        valueSource.setName(VALUE_SOURCE);
+        valueSource.setLabel("Source of Value");
+        valueSource.setHelpText("Tells the protocol mapper where to get the information. For now: COMPUTE or VC. Default is compute, in which this protocol mapper computes the current time in seconds.");
```

Comment:
```suggestion
        valueSource.setHelpText("Tells the protocol mapper where to get the information. For now: COMPUTE or VC. Default is compute, in which this protocol mapper computes the current time in seconds. With value `vc`, the time is computed from the verifiable credential issuance date.");
```

Reply:
Modified to read instead of computed.

Reply:
resolved with [cc0dff8](https://github.com/keycloak/keycloak/pull/31620/commits/cc0dff873da43ad6b33557fc66c189f25fd9273e)